### PR TITLE
#1642 cr popup same wbs

### LINKED ISSUE #1642: Other Change Requests from the Same WBS popup tab
### Current Features
branch off feature/Change-Request-Redesign for feature branch

image

### Desired Additional Features

We want a tab at the bottom of the page that shows the other change requests from the same WBS  Whether they submitted or reviewed those Change Requests.

Create a popup tab in Change Requests Details Page (similar to the drop down blocks)
Show all Change Requests that are for the same WBS as cards (same as in the overview page)

### Screenshots (as needed)

image

image
Epic https://github.com/Northeastern-Electric-Racing/FinishLine/issues/1645
branch off Change-Request-Redesign for feature branch
Message Zach Marino in slack for any design questions!
reference this and work together https://github.com/Northeastern-Electric-Racing/FinishLine/issues/1641

### PR DESCRIPTION
**Pull request by ryanhowe28**
## Changes

new popup tab for same wbs crs

## Screenshots

image
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1642

### CODE REVIEW

File: src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestDetailsView.tsx [hunk 1]
```diff
@@ -153,7 +153,9 @@ const ChangeRequestDetailsView: React.FC<ChangeRequestDetailsProps> = ({
       {deleteModalShow && (
         <DeleteChangeRequest modalShow={deleteModalShow} handleClose={handleDeleteClose} cr={changeRequest} />
       )}
-      <OtherChangeRequestsPopupTabs changeRequest={changeRequest} />
+      <Grid container spacing={1} style={{ display: 'flex' }}>
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestDetailsView.tsx:156]: I don't think this grid is doing anything

File: src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx [hunk 2]
```diff
@@ -15,10 +15,12 @@ import { fullNamePipe } from '../../utils/pipes';
 
 interface OtherChangeRequestsPopupTabsProps {
   changeRequest: ChangeRequest;
+  fromSubmitter: boolean;
 }
 
 const OtherChangeRequestsPopupTabs: React.FC<OtherChangeRequestsPopupTabsProps> = ({
-  changeRequest
+  changeRequest,
+  fromSubmitter
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx:23]: this prop is unused

File: src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx [hunk 3]
```diff
@@ -38,6 +40,8 @@ const OtherChangeRequestsPopupTabs: React.FC<OtherChangeRequestsPopupTabsProps>
       return b.dateSubmitted.getTime() - a.dateSubmitted.getTime();
     });
 
+  const crsFromWbs = changeRequests?.filter((cr) => cr.wbsName === changeRequest.wbsName);
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx:43]: can you sort these the same way the other one is being sorted

File: src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx [hunk 4]
```diff
@@ -113,9 +117,12 @@ const OtherChangeRequestsPopupTabs: React.FC<OtherChangeRequestsPopupTabsProps>
           mb: '-1px'
         }}
       >
-        {displayTab(1, `Other CR's from ${fullNamePipe(changeRequest.submitter)}`)}
+        {displayTab(1, `Other CR's from ${wbsPipe(changeRequest.wbsNum)}`)}
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx:120]: 'Other CR's on wbsNum'

**Review by RChandler234 — CHANGES_REQUESTED** [review]
I would say remove the arrows, bc this case where one is up and one is down looks a bit weird:
image

File: src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestDetailsView.tsx [hunk 5]
```diff
@@ -153,7 +153,7 @@ const ChangeRequestDetailsView: React.FC<ChangeRequestDetailsProps> = ({
       {deleteModalShow && (
         <DeleteChangeRequest modalShow={deleteModalShow} handleClose={handleDeleteClose} cr={changeRequest} />
       )}
-      <OtherChangeRequestsPopupTabs changeRequest={changeRequest} />
+      <OtherChangeRequestsPopupTabs changeRequest={changeRequest} fromSubmitter={true} />
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/ChangeRequestDetailsView.tsx:156]: can you delete the fromSubmitter prop here and remove it from the prop type

File: src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx [hunk 6]
```diff
@@ -113,9 +109,12 @@ const OtherChangeRequestsPopupTabs: React.FC<OtherChangeRequestsPopupTabsProps>
           mb: '-1px'
         }}
       >
-        {displayTab(1, `Other CR's from ${fullNamePipe(changeRequest.submitter)}`)}
+        {displayTab(1, `Other CR's on ${wbsPipe(changeRequest.wbsNum)}`)}
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx:112]: can you add the wbsName here like how it looks inside the cards that pop up when u click the tab

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx (hunk 6 above)
Comment (ryanhowe28) [reply src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx:112]: it looks wacky, needs to be on two lines for that to work

Comment (ryanhowe28) [reply src/frontend/src/pages/ChangeRequestDetailPage/OtherChangeRequestsPopupTabs.tsx:112]: image

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**